update-ubuntu-or-debian-server/run.sh

```
@@ -2,193 +2,915 @@
+        else
+            log "Continuing in unattended mode despite health issues" "WARNING"
+        fi
+    else
+        print_success "System health check passed"
+    fi
+}
+
+# Function to create backup
+create_backup() {
+    if [ "${backup_before_upgrade:-false}" = "true" ]; then
+        local backup_dir="/var/backups/bigbear-$(date +%Y%m%d-%H%M%S)"
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)